lib/engine/route.rb

```diff
@@ -342,7 +347,12 @@ def subsidy
     end
 
     def revenue_str
-      @revenue_str ||= @game.revenue_str(self)
+      @revenue_str ||=
+        if @route_error
+          visited_stops.map { |s| s.hex.name }.join('-')
+        else
+          @game.revenue_str(self)
+        end
```

Comment:
So,  is this needed any more?

Reply:
Probably not. Since you mentioned the problem where something like this happened in 18NewEngland, I was hoping this would be a catch-all but since we don't have that game to test any more, it's probably best to just stick with the fix that is needed for 1822CA.